### PR TITLE
Fix links and improve docu

### PR DESCRIPTION
**Pull request by bszwarc**
**Description**

Changes proposed in this pull request:

- Fix a broken link to examples
- Improve documentation for `provision` and `install` modules

### CODE REVIEW

File: install/README.md [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
 
 
 ## Usage
 
-The installation process consist of two phases:
-- `PrepareInstallation` - in this phase Tiller is installed and the Kyma Installer is deployed to the cluster along with the default configuration.
-- `StartInstallation` - in this phase the Installation is triggered by labeling Installation Custom Resource.
+The installation process consist of two phases triggered by the following functions: 
+* `PrepareInstallation` which creates all necessary Kyma resources, installs  Tiller, and deploys the Kyma Installer to the cluster along with the default configuration.
+* `StartInstallation` which  triggers the installation by labeling the Installation Custom Resource.
```
Comment (kazydek) [inline install/README.md]:
```suggestion
* `StartInstallation` which triggers the installation by labeling the Installation Custom Resource. 
```

File: README.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 Hydroform is an infrastructure SDK you can use to create and manage Kubernetes clusters. Hydroform allows you to manage your clusters on the desired target provider and location.
 
-Currently, Hydroform consists of two modules:
-- [Provision](./provision)
-- [Install](./install)
+Hydroform consists of two modules:
+
+- [Provision](./provision) which you can use to create and manage clusters.
```
Comment (kazydek) [inline README.md]:
```suggestion
- [Provision](./provision) which creates and manages clusters.
```

File: README.md [hunk 3]
```diff
@@ -4,6 +4,7 @@
...
+- [Install](./install) which you can use to install Kyma on clusters.
```
Comment (kazydek) [inline README.md]:
```suggestion
- [Install](./install) which installs Kyma on clusters.
```

File: provision/README.md [hunk 4]
```diff
@@ -2,11 +2,11 @@
 
 ## Overview
 
-The `hydroform` package contains the library used to create and manage clusters.
+The [`provision`](https://godoc.org/github.com/kyma-incubator/hydroform/provision) package provided in this module allows you to create and manage clusters.
 
 ## Usage
 
-Hydroform is a [Go package](https://godoc.org/github.com/kyma-incubator/hydroform) that you can use with any program. It gives you the following commands: `provision`, `status`, `credentials`, and `deprovision`. Use them to:
+The package includes the  `provision`, `status`, `credentials`, and `deprovision` commands. Use them to:
```
Comment (clebs) [inline provision/README.md]: These are not commands, but functions.

Comment (bszwarc) [reply provision/README.md]: Got it, changed :)

**Pull request merged**